gdpr/models.py

```diff
@@ -30,18 +30,32 @@ def create_from_purpose_slug(self, purpose_slug, source_object, issued_at=None,
             raise KeyError('Purpose with slug {} does not exits'.format(purpose_slug))
 
     def exists_valid_consent(self, purpose_slug, source_object):
-        return LegalReason.objects.filter(
-            source_object_content_type=ContentType.objects.get_for_model(source_object.__class__),
-            source_object_id=str(source_object.pk),
-            purpose_slug=purpose_slug,
-            is_active=True,
-            expires_at__gte=timezone.now(),
-        ).exists()
+        return LegalReason.objects.filter_source_instance_active_non_expired(
+            source_object).filter(purpose_slug=purpose_slug).exists()
+
+
+class LegalReasonQuerySet(models.QuerySet):
+
+    def filter_active(self):
```

Comment:
Nevím jestli má cenu kvůli jednomu boolean flagu dělat nový filtr...

Reply:
Kvůli samostatnému booleanu asi ne, ale viděl jsem to tak, že by se měl používat ten filtr a pak by se mu beztrastně mohla měnit logika.

Reply:
Ondra má pravdu je tu moc filterů u kterých si neumím představit, že by se nějak měnily. Nějak nevidím smysl

Reply:
OK, filter_active jsem odebral, ale ostatní zůstanou - ani trochu se mi nechce riskovat, že třeba do porovnání aktivity někdo použije jinou metodu získání času (timezone aware / non -aware).